src/gui/folder.cpp

```diff
@@ -640,7 +638,8 @@ void Folder::slotWatchedPathsChanged(const QSet<QString> &paths, ChangeReason re
     bool needSync = false;
     for (const auto &path : paths) {
         Q_ASSERT(FileSystem::isChildPathOf(path, this->path()));
```

Comment:
Because of being a slot anybody could call this with  paths not being children of this. pfff